tests/filecheck/dialects/x86/x86_assembly_emission.mlir

```diff
@@ -148,7 +148,7 @@ x86.directive ".align" "2"
 x86.label "label"
 // CHECK-NEXT: label:
 
-func.func @funcyasm() {
+x86_func.func @funcyasm() {
```

Comment:
the previous x86 assembly printing had a special handling for func, the only functional change to this PR is that now we require func to be lowered to x86_func before assembly printing.